_pytest/config.py

```diff
@@ -1041,15 +1041,21 @@ def _mark_plugins_for_rewrite(self, hook):
             for entry in entrypoint.dist._get_metadata(metadata)
         )
 
+        for name in self.iter_rewritable_modules(package_files):
+            hook.mark_rewrite(name)
+
+    def iter_rewritable_modules(self, package_files):
+        modules = []
         for fn in package_files:
-            is_simple_module = os.sep not in fn and fn.endswith('.py')
-            is_package = fn.count(os.sep) == 1 and fn.endswith('__init__.py')
+            is_simple_module = '/' not in fn and fn.endswith('.py')
+            is_package = fn.count('/') == 1 and fn.endswith('__init__.py')
             if is_simple_module:
-                module_name, ext = os.path.splitext(fn)
-                hook.mark_rewrite(module_name)
+                module_name, _ = os.path.splitext(fn)
+                modules.append(module_name)
```

Comment:
Please make this an actual generator instead of returning a list, replacing `modules.append(module_name)` by `yield module_name`.